sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/Generated/ConfidentialLedgerClient.cs

```diff
@@ -36,6 +36,14 @@ protected ConfidentialLedgerClient()
         {
         }
 
+        /// <summary> Initializes a new instance of ConfidentialLedgerClient. </summary>
+        /// <param name="ledgerUri"> The Confidential Ledger URL, for example https://contoso.confidentialledger.azure.com. </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="ledgerUri"/> or <paramref name="credential"/> is null. </exception>
+        public ConfidentialLedgerClient(Uri ledgerUri, TokenCredential credential) : this(ledgerUri, credential, new ConfidentialLedgerClientOptions())
```

Comment:
It looks like the constructor that requires ClientOptions is missing from the generated file.  The manual layer code may need to be updated to achieve the correct API pattern.

Reply:
This one is actually ok - the options ctor is customized and lives [here](https://github.com/Azure/azure-sdk-for-net/blob/1c90cb1fa2ff8d43983fee165ab2edbda8e036ea/sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/ConfidentialLedgerClient.cs#L21)

`ConfidentialLedgerIdentityServiceClient` is another story and will comment there for this.

Reply:
I think because the ClientOptions is optional in the custom constructor, you would have an ambiguous call site.  The fix is to make the ClientOptions required in the custom constructor.

Reply:
Good point - this would be an issue for callers that specify no options.